### PR TITLE
Simplifications

### LINKED ISSUE #1584: Clean up FC
**Instructions**
**User Story**
As a developer I want to have the FC library cleaned up in order to get rid of broken/obsolete/unused code and reduce maintenance effort.

Much of the code in FC predates C++-11 and modern Boost. Both C++-11 and Boost have standardized or at least well-proven replacements for much of FC. We should get rid of all code in FC that is either unused by the core, or for which standard and/or Boost replacements exist.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [ ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [x] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [x] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [x] Other (please add below)
Code quality

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

### PR DESCRIPTION
**Pull request by pmconrad**
Resolves #1584 and #998
Requires additional changes in FC: https://github.com/bitshares/bitshares-fc/pull/137

Measured 8% speedup for replaying 33M blocks

### CODE REVIEW

**Comment by pmconrad** [comment]
Snapshot comparison at 38M produces empty diff.

**Comment by abitmore** [comment]
Good job. Looks good so far.

**Comment by pmconrad** [comment]
Rebase to avoid conflict

**Comment by pmconrad** [comment]
Rebased.
Apparently Mac and Windows builds require additional changes, see https://github.com/bitshares/bitshares-fc/pull/140

File: libraries/chain/db_maint.cpp [hunk 1]
```diff
@@ -442,7 +440,7 @@ void database::initialize_budget_record( fc::time_point_sec now, budget_record&
    budget_u128 += ((uint64_t(1) << GRAPHENE_CORE_ASSET_CYCLE_RATE_BITS) - 1);
    budget_u128 >>= GRAPHENE_CORE_ASSET_CYCLE_RATE_BITS;
    if( budget_u128 < reserve.value )
```
Comment (jmjatlanta) [inline libraries/chain/db_maint.cpp]: This generates a signed/unsigned comparison warning on gcc 7.4.0, not the fault of this PR though.

File: libraries/chain/db_maint.cpp [hunk 2]
```diff
@@ -497,7 +495,7 @@ void database::process_budget()
       if( worker_budget_u128 >= available_funds.value )
```
Comment (jmjatlanta) [inline libraries/chain/db_maint.cpp]: This generates a signed/unsigned comparison warning on gcc 7.4.0, not the fault of this PR though.

**Pull request merged**